routers/api/v1/api.go

```diff
@@ -936,6 +936,14 @@ func Routes(ctx gocontext.Context) *web.Route {
 							Get(repo.GetIssueReactions).
 							Post(reqToken(), bind(api.EditReactionOption{}), repo.PostIssueReaction).
 							Delete(reqToken(), bind(api.EditReactionOption{}), repo.DeleteIssueReaction)
+						m.Combo("/dependencies").
+							Get(repo.GetIssueDependencies).
+							Post(reqRepoWriter(unit.TypeIssues, unit.TypePullRequests), bind(api.IssueMeta{}), repo.CreateIssueDependency).
```

Comment:
I think we have to check the permission in the `CreateIssueDependency` because we need to check write permission according if the issue is a pull request.